packages/oauth/oauth_server.js

```diff
@@ -135,7 +135,41 @@ OAuth._checkRedirectUrlOrigin = redirectUrl => {
   );
 };
 
-const middleware = (req, res, next) => {
+// Parses request payload to use in the handler
+//
+// @param {Object} req incoming oauth request
+//
+const parseRequestData = req => new Promise((resolve, reject) => {
```

Comment:
This entire thing should best be handled by existing middleware packages, since there's a lot of edge-cases and possible security concerns handled by packages like body-parser.

And some additional advantages like if the body is already parsed by another middleware you won't parse twice (e.g. the popular `simple:rest` package adds body parser middlewares for every route) and the oauth middleware stays synchronous, doing away with some of my other comments.

You'll want to make sure to add the middleware only for the oauth routes, so do something like:
`
WebApp.connectHandlers.use('/_oauth', bodyparser.json());
WebApp.connectHandlers.use('/_oauth', bodyparser.urlencoded());
`
before adding the main oauth middleware handler.

Reply:
[My first hack](https://github.com/meteor/meteor/pull/10751/commits/54d826f815bc820309ac791a16ed201d08eb6b48) did use body-parser but then I read somewhere to refrain from including external libs.

In rooch84 original issue, #10491. Ben talks about using a promise and so I thought it would be ok.

I'll change it if you think it should use bodyparser. As long as we gain the ability to parse post body data.

Reply:
For external libs we have to be careful which to include, especially on the client-side.
However, this is server-side only and body-parser is a very well-known library. Otherwise this implementation would at least need some DOS protection by limiting the body size and limiting the parameters. Maybe for some other oauth provider it would need to inflate the data first etc. All of this is already taken care of by body-parser. But maybe let's wait for some input from @benjamn before you make the change.

Reply:
hi @jramer are you going to use bodyparser or is the PR ready?

Reply:
I'm waiting for input. It's your choice. 
I've been using this in production with google/facebook/apple oauth.

Reply:
If everything works when using `body-parser`, I think this is a case where an external library with 12MM weekly downloads is preferable to a handwritten reimplementation, for the reasons @sebakerckhof mentioned.

I am excited to see this PR get merged, so please go ahead and reinstate `body-parser` (unless there's a specific reason it won't work, in which case we can talk about that).

Reply:
Cool. I’ll try to fix it ASAP

Reply:
I think it's ready, and better than before. 🎉